draft-ietf-jose-hpke-encrypt.md

```diff
@@ -106,167 +109,222 @@ informative:
 
-## Encapsulated Keys {#encapsulated-keys}
+## Encapsulated Secrets {#encapsulated-secrets}
 
-HPKE encapsulated key is defined in {{Section 5 of I-D.ietf-hpke-hpke}}.
+HPKE encapsulated secret is defined in {{Section 5 of I-D.ietf-hpke-hpke}}.
 
-In HPKE JWE Integrated Encryption, the JWE Encrypted Key of the sole recipient is the HPKE encapsulated key.
+When using Integrated Encryption, the JWE Encrypted Key of the sole recipient
+is the HPKE encapsulated secret.
 
-In HPKE JWE Key Encryption, each recipient JWE Encrypted Key is the encrypted content encryption key, and the value of JOSE Header parameter "ek"
-is base64url-encoded HPKE encapsulated key.
+When using Key Encryption, each recipient's JWE Encrypted Key
+is the encrypted content encryption key, and the value of header parameter "ek"
+is the base64url encoding of the HPKE encapsulated secret.
 
 # Integrated Encryption
 
-In HPKE JWE Integrated Encryption:
+When using Integrated Encryption:
 
-- The protected header MUST contain an "alg" that is JOSE-HPKE algorithm.
-- The protected header MUST contain an "enc" with value "int". This is an explicit exception to requirement in {{Section 4.1.2 of RFC7516}} that
-"enc" must be an AEAD algorithm. This is appropriate, as HPKE will perform plaintext encryption.
-- The protected header parameters "psk_id" MAY be present.
+- The protected header MUST contain an "alg" value that uses Integrated Encryption.
+- The "enc" header parameter MUST NOT be present.
+  This is because no separate content encryption algorithm is used in this mode.
+- The protected header parameter "psk_id" MAY be present.
 - The protected header parameter "ek" MUST NOT be present.
 - There MUST be exactly one recipient.
-- The JWE Encrypted Key MUST be encapsulated key, as defined in {{Section 5 of I-D.ietf-hpke-hpke}}.
+- The JWE Encrypted Key MUST be the encapsulated secret, as defined in {{Section 5 of I-D.ietf-hpke-hpke}}.
 - The JWE Initialization Vector and JWE Authentication Tag MUST be the empty octet sequence.
 - The JWE AAD MAY be present when using the JWE JSON Serialization.
-- The JWE Ciphertext is the ciphertext defined in {{Section 5.2 of I-D.ietf-hpke-hpke}}.
-- The HPKE info parameter contains the encoding of the Recipient_structure, which is described in
-  {{recipient_structure}}.
-- The HPKE aad parameter MUST be set to the "Additional Authenticated Data encryption parameter", as specified in Step 14 of {{Section 5.1 of RFC7516}}.
-- Then follow Steps 11-19 of {{Section 5.1 of RFC7516}} (Message Encryption).
+- The HPKE aad parameter MUST be set to the "Additional Authenticated Data encryption parameter" value specified in Step 15 of {{encryption}}.
+- The HPKE info parameter defaults to the empty string;
+  mutually known private information (a concept also utilized in {{NIST.SP.800-56Ar3}})
+  MAY be used instead so the application can include it during key derivation.
+externally provided information MAY be used instead.
```

Comment:
```suggestion
- The HPKE info parameter defaults to the empty string;
  mutually known private information (a concept also utilized in {{NIST.SP.800-56Ar3}})
  MAY be used instead so the application can include it during key derivation.
```

Reply:
Using both info and aad simultaneously would violate the API guidance in {{Section 8.1 of I-D.ietf-hpke-hpke}}, which advises against the use of both parameters in single-shot HPKE APIs.